airflow/models.py

```diff
@@ -1598,11 +1606,11 @@ def handle_failure(self, error, test_mode=False, context=None, session=None):
 
         # Let's go deeper
         try:
-            # try_number is incremented by 1 during task instance run. So the
+            # _try_number is incremented by 1 during task instance run. So the
             # current task instance try_number is the try_number for the next
             # task instance run. We only mark task instance as FAILED if the
             # next task instance try_number exceeds the max_tries.
-            if task.retries and self.try_number <= self.max_tries:
+            if task.retries and self._try_number <= self.max_tries:
```

Comment:
I think it should be possible to use `try_number` here

Reply:
Oh yes, cos this fn is only called when state == RUNNING.